Sources/JobsPostgres/PostgresJobsQueue.swift

```diff
@@ -158,11 +158,12 @@ public final class PostgresJobQueue: JobQueueDriver {
     /// shutdown queue once all active jobs have been processed
     public func shutdownGracefully() async {}
 
-    func popFirst() async throws -> QueuedJob<JobID>? {
+    func popFirst() async throws -> Result<QueuedJob<JobID>?, Error> {
```

Comment:
It seems a little odd to have a throwing function return a `Result` ie it has two ways to report errors. I would still have it return the QueuedJob<JobID>, but at the bottom of the function return `try result.get()`

```swift
let result = try await self.client.withTransaction {
    ...
}
return try result.get()
```